dataprofiler/profilers/profile_builder.py

```diff
@@ -2512,6 +2522,11 @@ def tqdm(level: Set[int]) -> Generator[int, None, None]:
                     min_true_samples = self._profile[prof_idx]._min_true_samples
                 try:
                     null_values = self._profile[prof_idx]._null_values
```

Comment:
This is a huge add and most LGTM, one big here though bc a doctor is mutable, this will change self._null_values with the update. If we instead copy prior to a variable, that would alleviate the issue. Great job though!

Reply:
Just need to fix in the locations where we update

Reply:
@tonywu315

Reply:
added `.copy()`